forum_leaderboard/forum_leaderboard.py

```diff
@@ -0,0 +1,111 @@
+    )
+    # Studio view not yet written. To test, set the default here,
+    # or manually set in Mongo.
+    discussion_id = String(
+        default="", scope=Scope.settings,
+        help="The ID of the inline discussion to tally leading threads for.",
+    )
+    count = Integer(
+        default=10, scope=Scope.settings,
+        help="How many threads to display."
+    )
+
+    def resource_string(self, path):
+        """
+        Handy helper for getting resources from our kit.
+        """
+        data = pkg_resources.resource_string(__name__, path)
+        return data.decode("utf8")
+
+    def create_fragment(self, html, context=None, css=None, javascript=None, initialize=None):
+        """
+        Create an XBlock, given an HTML resource string and an optional context, list of CSS
+        resource strings, list of JavaScript resource strings, and initialization function name.
+        """
+        html = Template(self.resource_string(html))
+        context = context or {}
+        css = css or []
+        javascript = javascript or []
+        frag = Fragment(html.render(Context(context)))
+        for sheet in css:
+            frag.add_css(self.resource_string(sheet))
+        for script in javascript:
+            frag.add_javascript(self.resource_string(script))
+        if initialize:
+            frag.initialize_js(initialize)
+        return frag
+
+    def student_view(self, context=None):
+        """ Renders student view for LMS and Studio """
+        if hasattr(self, 'xmodule_runtime') and getattr(self.xmodule_runtime, 'is_author_mode', False):
+            fragment = self._student_view_studio()
+        else:
+            fragment = self._student_view_lms()
+
+        return fragment
+
+    def _student_view_lms(self):
+        """
+        The primary view of the ForumLeaderboardXBlock, shown to students
+        when viewing courses.
+        """
+        discussion_service = self.xmodule_runtime.service(self, 'discussion')
+        discussion_context = discussion_service.get_inline_template_context()
+        course = discussion_context['course']
+        threads = cc.Thread.search({
+            'course_id': unicode(course.location.course_key), 'commentable_id': self.discussion_id,
+            'sort_key': 'votes', 'per_page': self.count
+        })[0]
+
+        context = {
+            'threads': threads,
+            'display_name': self.display_name
+        }
+        return self.create_fragment(
+            "static/html/forum_leaderboard.html", context=context,
+            css=["static/css/forum_leaderboard.css"],
+            javascript=["static/js/src/forum_leaderboard.js"],
+            initialize='ForumLeaderboardXBlock')
+
+    def _student_view_studio(self):
+        return self.create_fragment(
+            "static/html/forum_leaderboard_studio.html",
+            context={'discussion_id': self.discussion_id},
+            css=["static/css/forum_leaderboard.css"])
```

Comment:
@bradenmacdonald It feels dirty, but I think I'll just put the import statements for the validator models within that function, since it should never be called by the SDK anyway (I don't think). That will allow me to use them without sabotaging myself later.